modules/core/src/main/java/org/mycontroller/standalone/gateway/philipshue/PhilipsHueGatewayPoller.java

```diff
@@ -0,0 +1,267 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mycontroller.standalone.gateway.philipshue;
+
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.mycontroller.restclient.core.ClientResponse;
+import org.mycontroller.restclient.philips.hue.PhilipsHueClient;
+import org.mycontroller.restclient.philips.hue.PhilipsHueClientBuilder;
+import org.mycontroller.restclient.philips.hue.model.LightState;
+import org.mycontroller.restclient.philips.hue.model.State;
+import org.mycontroller.standalone.gateway.model.GatewayPhilipsHue;
+import org.mycontroller.standalone.message.McMessageUtils.MESSAGE_TYPE;
+import org.mycontroller.standalone.message.McMessageUtils.MESSAGE_TYPE_INTERNAL;
+import org.mycontroller.standalone.message.McMessageUtils.MESSAGE_TYPE_SET_REQ;
+import org.mycontroller.standalone.message.RawMessage;
+import org.mycontroller.standalone.message.RawMessageQueue;
+import org.mycontroller.standalone.model.philips.Color;
+import org.mycontroller.standalone.model.philips.PHUtilities;
+import org.mycontroller.standalone.provider.philipshue.PhilipsHueUtils;
+import org.mycontroller.standalone.utils.McUtils;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Fraid(https://github.com/Fraid)
+ */
+@Slf4j
+public class PhilipsHueGatewayPoller implements Runnable {
+    private boolean terminate = false;
+    private boolean terminated = false;
+    private PhilipsHueClient philipsHueClient;
+    private boolean onRequestUpdate = false;
+
+    public PhilipsHueGatewayPoller() {
+    }
+
+    private GatewayPhilipsHue gateway = null;
+
+    public PhilipsHueGatewayPoller(GatewayPhilipsHue gateway) throws Exception {
+        this.gateway = gateway;
+        philipsHueClient = new PhilipsHueClientBuilder().uri(gateway.getUrl())
+                .addProperty(PhilipsHueClient.KEY_AUTHORIZED_USER, gateway.getAuthorizedUser()).build();
+
+    }
+
+    @Override
+    public void run() {
+        //Initial delay, allow to add this object in McObject manager
+        try {
+            Thread.sleep(McUtils.SECOND * 10);
+        } catch (InterruptedException ex) {
+            _logger.error("Exception, ", ex);
+        }
+        while (!isTerminate()) {
+            try {
+                _logger.debug("Getting hue lights...on user request : {} ", onRequestUpdate);
+                final ClientResponse<Map<String, LightState>> listAll = philipsHueClient.lights().listAll();
```

Comment:
@Fraid We may go with `listAll` only once, then we have to call `listNew`.
If we call `listAll` every time, will update forcefully with existing status and always shows `last seen` as a very recent timestamp.